packages/amplify_api/test/amplify_api_graphql_helpers_test.dart

```diff
@@ -46,4 +57,50 @@ void main() {
     }
     fail("Expected an ApiException");
   });
+
+  test('Query returns a GraphQLRequest<Blog> when provided a modelType',
+      () async {
+    String id = UUID.getUUID();
+    GraphQLRequest<Blog> req = ModelQueries.get<Blog>(Blog.classType, id);
+    List<GraphQLResponseError> errors = [];
+    String data = '''{
+        "getBlog": {
+            "createdAt": "2021-01-01T01:00:00.000000000Z",
+            "id": "$id",
+            "name": "TestAppBlog"
+        }
+    }''';
+
+    GraphQLResponse<Blog> response = GraphQLResponseDecoder.instance
+        .decode<Blog>(request: req, data: data, errors: errors);
+
+    expect(response.data.runtimeType, Blog);
```

Comment:
```dart
expect(response.data, isA<Blog>());
```

This is guaranteed by the type contract, though, so it will always be true. What we want to test here is probably that is was decoded properly, so you can create the Blog object it's supposed to be and compare those, instead.

Reply:
Thank you!! I could not find how to do this on google, much better 🥳